generate/generate.go

```diff
@@ -594,11 +606,18 @@ func writeGeneratedCode(target string, code string) error {
 }
 
 func checkFileCanBeOverwritten(path string) error {
-	_, err := loadGeneratedCode(path)
+	_, _, err := loadGeneratedCode(path)
 	return err
 }
 
-func loadGeneratedCode(path string) ([]byte, error) {
+// loadGeneratedCode will load the generated code at the given path.
+// It returns an error if it can't read the file of if the file is not
+// a Terramate generated file.
+//
+// The returned boolean indicates if the file exists, so the contents of
+// the file + true is returned if a file is found, but if no file is found
+// it will return an empty string and false indicating that the file doesn't exist.
+func loadGeneratedCode(path string) (string, bool, error) {
```

Comment:
Could just use the empty string as also meaning "file dont exist" since we never generate empty files on terramate + empty files wont be detected as being generated by terramate. But I usually dislike overloading the meaning of something ("" means file exists and is empty AND file doesn't exist)...so this seemed better even though not strictly necessary.